toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -352,6 +371,136 @@ func findPartition(mountIdType imagecustomizerapi.MountIdentifierType, mountId s
 	return partition, partitionIndex, nil
 }
 
+func findExtendedPartition(mountIdType imagecustomizerapi.ExtendedMountIdentifierType, mountId string,
+	partitions []diskutils.PartitionInfo, buildDir string,
+) (diskutils.PartitionInfo, int, error) {
+	matchedPartitionIndexes := []int(nil)
+	for i, partition := range partitions {
+		matches := false
+		switch mountIdType {
+		case imagecustomizerapi.ExtendedMountIdentifierTypeUuid:
+			matches = partition.Uuid == mountId
+		case imagecustomizerapi.ExtendedMountIdentifierTypePartUuid:
+			matches = partition.PartUuid == mountId
+		case imagecustomizerapi.ExtendedMountIdentifierTypePartLabel:
+			matches = partition.PartLabel == mountId
+		case imagecustomizerapi.ExtendedMountIdentifierTypeDev:
+			matches, err := findExtendedDevPartition(partition, partitions, buildDir)
+			if err != nil {
+				return diskutils.PartitionInfo{}, 0, err
+			}
+			if matches {
+				matchedPartitionIndexes = append(matchedPartitionIndexes, i)
+			}
+		}
+		if matches {
+			matchedPartitionIndexes = append(matchedPartitionIndexes, i)
+		}
+	}
+
+	if len(matchedPartitionIndexes) < 1 {
+		err := fmt.Errorf("partition not found (%s=%s)", mountIdType, mountId)
+		return diskutils.PartitionInfo{}, 0, err
+	}
+	if len(matchedPartitionIndexes) > 1 {
+		err := fmt.Errorf("too many matches for partition found (%s=%s)", mountIdType, mountId)
+		return diskutils.PartitionInfo{}, 0, err
+	}
+
+	partitionIndex := matchedPartitionIndexes[0]
+	partition := partitions[partitionIndex]
+
+	return partition, partitionIndex, nil
+}
+
+func findExtendedDevPartition(partition diskutils.PartitionInfo, partitions []diskutils.PartitionInfo,
```

Comment:
This function seems to be specific to finding verity partition (uki/non-uki ends up calling `checkVerityRootPartUUID`), should we rename to indicate its purpose? I'm not sure why it's named `ExtendedDevPartition`?)